clang osx warning in `find_all_idxs_by`(search.hpp)

(Sorry to drown you with messages today :-)

With apple-clang under osx I have a warning inside search.hpp (which is considered as an error during the build).

````
clang --version
Apple clang version 12.0.0 (clang-1200.0.32.21)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
````

The file `transform_test.cpp` will fail to compile at this line:
````cpp
TEST_CASE("transform_test, keep_if_parallelly")
{
    const std::vector<int> v = {1, 2, 3, 2, 4, 5};
-->    auto result = fplus::keep_if_parallelly(is_even, v);
    REQUIRE_EQ(result, std::vector<int>({2, 2, 4}));
}
````

And the reported warning is inside `find_all_idxs_by` at the line `for (const auto& x : xs)`:

````cpp
template <typename ContainerOut = std::vector<std::size_t>,
        typename UnaryPredicate, typename Container>
ContainerOut find_all_idxs_by(UnaryPredicate p, const Container& xs)
{
    internal::check_unary_predicate_for_container<UnaryPredicate, Container>();
    std::size_t idx = 0;
    ContainerOut result;
    auto itOut = internal::get_back_inserter(result);
    --> for (const auto& x : xs)
    {
        if (internal::invoke(p, x))
            *itOut = idx;
        ++idx;
    }
    return result;
}

````

Here is the complete compilation log:
````cpp
CMakeFiles/transform_test.dir/transform_test.cpp.o -c ../test/transform_test.cpp
In file included from ../test/transform_test.cpp:9:
In file included from ../include/fplus/fplus.hpp:12:
In file included from ../include/fplus/container_properties.hpp:14:
../include/fplus/search.hpp:117:22: error: loop variable 'x' is always a copy because the range of type 'const std::__1::vector<bool, std::__1::allocator<bool> >' does not return a reference [-Werror,-Wrange-loop-analysis]
    for (const auto& x : xs)
                     ^
../include/fplus/transform.hpp:403:23: note: in instantiation of function template specialization 'fplus::find_all_idxs_by<std::__1::vector<unsigned long, std::__1::allocator<unsigned long> >, bool (*)(const bool &), std::__1::vector<bool, std::__1::allocator<bool> > >' requested here
    const auto idxs = find_all_idxs_by(
                      ^
../test/transform_test.cpp:168:26: note: in instantiation of function template specialization 'fplus::keep_if_parallelly<bool (*)(int), std::__1::vector<int, std::__1::allocator<int> > >' requested here
    auto result = fplus::keep_if_parallelly(is_even, v);
                         ^
../include/fplus/search.hpp:117:10: note: use non-reference type 'std::__1::__bit_const_reference<std::__1::vector<bool, std::__1::allocator<bool> > >'
    for (const auto& x : xs)
         ^~~~~~~~~~~~~~~
1 error generated.
[23/82] Building CXX object test/CMakeFiles/container_common_test.dir/container_common_test.cpp.o
ninja: build stopped: subcommand failed.
````

I could silence i with a simple
````cpp
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wrange-loop-analysis"
````

but may be you have a better idea (but I'm afraid it might require some more SFINAE noise...)

--- a/include/fplus/transform.hpp
+++ b/include/fplus/transform.hpp
@@ -400,8 +400,12 @@ typename Container::value_type reduce_1_parallelly(F f, const Container& xs)
 template <typename Pred, typename Container>
 Container keep_if_parallelly(Pred pred, const Container& xs)
 {
+    // Avoid a temporary std::vector<bool>.
     const auto idxs = find_all_idxs_by(
-        identity<bool>, transform_parallelly(pred, xs));
+        is_equal_to<std::uint8_t>(1),
+        transform_parallelly([pred](const auto & x) -> std::uint8_t {
+            return pred(x) ? 1 : 0;
+        }, xs));
     return elems_at_idxs(idxs, xs);
 }
 

